### ISSUE TITLE
Cannot build keyboards on Mac

### ISSUE THREAD

**Issue by albbas** [divvun/kbdgen#20]
Tested with keyboard-fkv and keyboard-sme

Cloned this repo, built and installed it using the instructions in the readme-file. Cloned keyboard-fkv and keyboard-sme, ran `./autogen.sh&&./configure&&make mac RUST_BACKTRACE=full`

The output for keyboard-sme:

```
keyboard-sme on  main 
❯ ./autogen.sh&&./configure&&make mac RUST_BACKTRACE=full

Initial automake setup of keyboard-sme

configure.ac:35: installing 'build-aux/install-sh'
configure.ac:35: installing 'build-aux/missing'
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for kbdgen... $HOME/.cargo/bin/kbdgen
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
-- Building giellaltkeyboards-sme 20130628:

For more ./configure options, run ./configure --help

To build installation packages for the desktop keyboards, do:
    make

*** Building keyboards for the sme language. ***

CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh '$HOME/repos/giellalt/keyboard-sme/build-aux/missing' autoconf

*** macOS build! ***

  KBDGEN   mac
2024-02-19T10:20:22.073504Z  INFO kbdgen: kbdgen 3.0.0-beta.5
2024-02-19T10:20:22.073666Z  INFO kbdgen::bundle: Canonical Bundle Path: "$HOME/repos/giellalt/keyboard-sme/sme.kbdgen"
2024-02-19T10:20:22.076164Z  WARN kbdgen::bundle: Saw target with name chrome but did not parse
2024-02-19T10:20:22.080694Z  WARN kbdgen::bundle: Saw resource folder with name chrome but did not parse
thread 'main' panicked at src/build/macos/macos_bundle.rs:171:18:
convert failed to run: Os { code: 2, kind: NotFound, message: "No such file or directory" }
stack backtrace:
   0:        0x102af84e0 - std::backtrace_rs::backtrace::libunwind::trace::h815672f5996c9e34
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:        0x102af84e0 - std::backtrace_rs::backtrace::trace_unsynchronized::h2e094b2d8c270437
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:        0x102af84e0 - std::sys_common::backtrace::_print_fmt::hae24b71afea95695
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/sys_common/backtrace.rs:67:5
   3:        0x102af84e0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h6d4268b2ed62fb94
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/sys_common/backtrace.rs:44:22
   4:        0x102b1bc6c - core::fmt::rt::Argument::fmt::h835101b6de0df17f
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/fmt/rt.rs:138:9
   5:        0x102b1bc6c - core::fmt::write::h5d55d44549819258
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/fmt/mod.rs:1094:21
   6:        0x102af4330 - std::io::Write::write_fmt::hc515897f91abd6cf
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/io/mod.rs:1714:15
   7:        0x102af8320 - std::sys_common::backtrace::_print::h5844dab5cfa39fca
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/sys_common/backtrace.rs:47:5
   8:        0x102af8320 - std::sys_common::backtrace::print::h2c300c1ebedfc73c
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/sys_common/backtrace.rs:34:9
   9:        0x102af9b18 - std::panicking::default_hook::{{closure}}::h0aa9be5c44269370
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:270:22
  10:        0x102af9844 - std::panicking::default_hook::h2c0ef097934ee9e6
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:290:9
  11:        0x102afa060 - std::panicking::rust_panic_with_hook::h84c8637cb6e56008
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:707:13
  12:        0x102af9f6c - std::panicking::begin_panic_handler::{{closure}}::h25482adda06c7b7f
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:599:13
  13:        0x102af896c - std::sys_common::backtrace::__rust_end_short_backtrace::h0c6f3beb22324a29
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/sys_common/backtrace.rs:170:18
  14:        0x102af9cd0 - rust_begin_unwind
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:595:5
  15:        0x102b57e2c - core::panicking::panic_fmt::h9072a0246ecafd14
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/panicking.rs:67:14
  16:        0x102b58154 - core::result::unwrap_failed::hd7600d03a3086be4
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/result.rs:1652:5
  17:        0x1024f3a18 - kbdgen::build::macos::macos_bundle::MacOsBundle::write_all::h29cd13e171597baa
  18:        0x10254afa8 - <kbdgen::build::macos::generate_macos::GenerateMacOs as kbdgen::build::BuildStep>::build::{{closure}}::h222022dfac8ab497
  19:        0x1023db510 - kbdgen::build::BuildSteps::build_full::{{closure}}::h3477456453765d01
  20:        0x1023c220c - kbdgen::main::{{closure}}::h9f9e8e203baf7698
  21:        0x1023c11c8 - tokio::runtime::park::CachedParkThread::block_on::h794b375fba6b5b81
  22:        0x1023e1250 - tokio::runtime::context::runtime::enter_runtime::hbde36e1734f5e246
  23:        0x1023e8484 - tokio::runtime::runtime::Runtime::block_on::hd5bb35362f34833d
  24:        0x1023f33ec - kbdgen::main::heda98f19afe5bdee
  25:        0x1023e1a24 - std::sys_common::backtrace::__rust_begin_short_backtrace::h33a0ef5a051bf6db
  26:        0x1023e3b84 - std::rt::lang_start::{{closure}}::h2e9ed624cb725e34
  27:        0x102aed8a8 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h7fff7c88c61df16f
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/ops/function.rs:284:13
  28:        0x102aed8a8 - std::panicking::try::do_call::h5c3526811781b4da
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:502:40
  29:        0x102aed8a8 - std::panicking::try::h64cb573047f67390
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:466:19
  30:        0x102aed8a8 - std::panic::catch_unwind::h0ee316aa6f14634c
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panic.rs:142:14
  31:        0x102aed8a8 - std::rt::lang_start_internal::{{closure}}::hbd896cc316ba7338
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs:148:48
  32:        0x102aed8a8 - std::panicking::try::do_call::hbae6886803fae47e
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:502:40
  33:        0x102aed8a8 - std::panicking::try::hcc0fbb3995231d31
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:466:19
  34:        0x102aed8a8 - std::panic::catch_unwind::hf36acbaf210582be
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panic.rs:142:14
  35:        0x102aed8a8 - std::rt::lang_start_internal::hadaf077a6dd0140b
                               at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs:148:20
  36:        0x1023f34f4 - _main
make: *** [Makefile:603: mac] Error 101
```

**Comment by albbas** [comment]
"Fixed" with 83fab2d, document dependency on `imagemagick`

**Comment by snomos** [comment]
Now fixed one level up, by adding a test for `convert` in `configure.ac`, and erroring out if not found. Cf commit https://github.com/giellalt/template-keyboard-und/commit/4c943b609dddca98aa904e9be25e77177ceb8edd
